base/src/blockchain_state.rs

```diff
@@ -82,6 +84,56 @@ impl ToChunkedROInput for BlockchainState {
     }
 }
 
+impl Into<BlockchainState> for BlockchainStateLegacy {
```

Comment:
Why Into trait? I don't think they are compatible with those breaking changes.

Reply:
in some chain selection test cases we directly read the block json and check, this trait is introduced to keep those compatible with new ProtocolState